modules/aniviewBidAdapter.js

```diff
@@ -0,0 +1,199 @@
+import { VIDEO } from '../src/mediaTypes';
+import { registerBidder } from '../src/adapters/bidderFactory';
+import { Renderer } from '../src/Renderer';
+
+const BIDDER_CODE = 'aniview';
+const TTL = 600;
+
+function avRenderer(bid) {
+  bid.renderer.push(function() {
+    let eventCallback = bid && bid.renderer && bid.renderer.handleVideoEvent ? bid.renderer.handleVideoEvent : null;
+    window.aniviewRenderer.renderAd({
+      id: bid.adUnitCode + '_' + bid.adId,
+      debug: window.location.href.indexOf('pbjsDebug') >= 0,
+      placement: bid.adUnitCode,
+      width: bid.width,
+      height: bid.height,
+      vastUrl: bid.vastUrl,
+      vastXml: bid.vastXml,
+      config: bid.params[0].rendererConfig,
+      eventsCallback: eventCallback,
+      bid: bid
+    });
+  });
+}
+
+function newRenderer(bidRequest) {
+  const renderer = Renderer.install({
+    url: 'https://player.aniview.com/script/6.1/prebidRenderer.js',
+    config: {},
+    loaded: false,
+  });
+
+  try {
+    renderer.setRender(avRenderer);
+  } catch (err) {
+  }
+
+  return renderer;
+}
+
+function isBidRequestValid(bid) {
+  if (!bid.params || !bid.params.AV_PUBLISHERID || !bid.params.AV_CHANNELID) { return false; }
+
+  return true;
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  let bidRequests = [];
+
+  for (let i = 0; i < validBidRequests.length; i++) {
+    let bidRequest = validBidRequests[i];
+
+    if (!bidRequest.sizes || !bidRequest.sizes.length) {
```

Comment:
**bidRequest.sizes** may be deprecated in the next major release. Alternative is to use the new **bidRequest.mediaTypes** object.